[ADD] 스탬프 로컬 저장 기능 추가

JUNY0110 opened the pull request:
### 작업 내용
- 수집한 스탬프 정보 저장 기능(UserDefaults)
- EventsViewModel 추가

### 스크린샷
1. 스탬프 적립
2. 앱 종료 후 재실행
3. 스탬프 저장 여부 확인

|전|후|
|:-:|:-:|
|![ezgif com-video-to-gif-converter](https://github.com/user-attachments/assets/4d14340a-bf89-4a30-b9c5-5147b8f4d8d6)|![Video to GIF converter](https://github.com/user-attachments/assets/94f8e4e4-bcc0-4416-bddd-72ec8a36750f)|

JUNY0110 committed:
feat: Event 객체 타입 변환을 위한 Codable 프로토콜 채택

JUNY0110 committed:
feat: EventsViewModel 생성

- 스탬프 Save, Load 기능 메서드 추가

JUNY0110 committed:
feat: viewModel 적용

doitduri commented on LetSwift/View/Events/Views/EventsView.swift:
```diff
@@ -11,12 +11,17 @@ import SwiftUI
     // MARK: - Properties
-    @State private var eventCellStates: [Event] = []
     @Binding var payload: String
+    private var viewModel = EventsViewModel()
```
```suggestion
    private let viewModel = EventsViewModel()
```

`viewModel`이 런타임에서 굳이 변경되지 않은 값이면, `let`으로 선언하는 것은 어떠신가요!

JUNY0110 replied:
> `viewModel`이 런타임에서 굳이 변경되지 않은 값이면, `let`으로 선언하는 것은 어떠신가요!

311f359 
State를 제거하다가 놓쳤네요..! 꼼꼼한 확인 감사합니다!

doitduri commented on LetSwift/View/Events/ViewModels/EventsViewModel.swift:
```diff
@@ -0,0 +1,46 @@
+    }
+    
+    private func loadStamps() -> [Event] {
+        guard let savedEvents = UserDefaults.standard.object(forKey: "events") as? Data else { return [] }
```
기존에 `UserDefaults`를 관리하는 `UserDefaultsWrapper` class가 있어서 컨벤션을 위해 한번 옮겨주세요~!

JUNY0110 replied:
> 기존에 `UserDefaults`를 관리하는 `UserDefaultsWrapper` class가 있어서 컨벤션을 위해 한번 옮겨주세요~!

71af84d
적용했습니다!

doitduri commented on LetSwift/View/Events/Views/EventsView.swift:
```diff
@@ -71,10 +76,12 @@ fileprivate extension EventsView {
                     .overlay {
-                        if index < eventCellStates.count {
-                            Image(uiImage: eventCellStates[index].image)
+                        if index < viewModel.eventCellStates.count {
```
```suggestion
                        if viewModel.eventCellStates.indices.contains(index) {
```

viewModel에서 가지고 있는 model을 조회 하는 용도면 `contains` 사용하는게 더 직관적일 것 같아요-! 검토 부탁드려요~

JUNY0110 replied:
> viewModel에서 가지고 있는 model을 조회 하는 용도면 `contains` 사용하는게 더 직관적일 것 같아요-! 검토 부탁드려요~

698cb24
index를 사용하는 코드 맥락을 일치시킨다는 점에서 말씀하신 방향이 더 좋은 것 같습니다! 수정완료했습니다.

doitduri reviewed (commented):
@JUNY0110 님 PR 확인 했습니다. 소소한 코드 개선 건 있어서 댓글 달았어요~
한번 확인 부탁드려요 💯 💯 💯 💯  고생하셨어요 😄

JUNY0110 committed:
mod: 선언 타입 변경 (var-> let)

JUNY0110 committed:
style: 코드 스타일 수정

- index < eventCellStates.count를 eventCellStates.indices.contains(index)로 변경하여 코드 맥락의 일관성 유지

JUNY0110 committed:
mod: UserDefualtsManager 적용

- 기존 구성된 UserDefaultsWrapper 활용
